app/importer/server/classes/ImportDataConverter.ts

```diff
@@ -202,43 +214,48 @@ export class ImportDataConverter {
 		subset(userData.customFields, 'customFields');
 	}
 
-	updateUserId(_id: string, userData: IImportUser): void {
+	updateUser(existingUser: IUser, userData: IImportUser): void {
+		const { _id } = existingUser;
+
+		userData._id = _id;
+
 		// #ToDo: #TODO: Move this to the model class
 		const updateData: Record<string, any> = {
 			$set: {
 				roles: userData.roles || ['user'],
 				type: userData.type || 'user',
 				...userData.statusText && { statusText: userData.statusText },
 				...userData.bio && { bio: userData.bio },
-				...userData.name && { name: userData.name },
 				...userData.services?.ldap && { ldap: true },
 			},
 		};
 
 		this.addCustomFields(updateData, userData);
 		this.addUserServices(updateData, userData);
 		this.addUserImportId(updateData, userData);
-		this.flagEmailsAsVerified(updateData, userData);
+		this.addUserEmails(updateData, userData, existingUser.emails || []);
 		Users.update({ _id }, updateData);
-	}
-
-	updateUser(existingUser: IUser, userData: IImportUser): void {
-		userData._id = existingUser._id;
 
-		this.updateUserId(userData._id, userData);
+		if (userData.utcOffset) {
+			Users.setUtcOffset(_id, userData.utcOffset);
+		}
 
-		if (userData.importIds.length) {
-			this.addUserToCache(userData.importIds[0], existingUser._id, existingUser.username);
+		if (userData.name || userData.username) {
+			saveUserIdentity({ _id, name: userData.name, username: userData.username, joinDefaultChannelsSilenced: true });
```

Comment:
the user should be added to default channels only if it's being added, that's the default behavior everywhere else. as I mentioned in other comment, pls don't add this behavior to `saveUserIdentity`..

if you need to add the user to default channels call `addUserToDefaultChannels` directly. probably better call it from `insertUser`.